### ISSUE TITLE
No implementations provided for Cohttp modules

### ISSUE THREAD

**Issue by Thegaram** [mirage/ocaml-cohttp#604]
I am trying to build `server_example.ml` from the README on a Linux machine running inside a Virtualbox VM on Xen, but I am getting `No implementation provided` errors for Cohttp modules.

```
$ ocaml --version
The OCaml toplevel, version 4.06.1

$ opam --version
1.2.2

$ opam list --installed | grep cohttp
cohttp                        1.0.2  An OCaml library for HTTP clients and serve
cohttp-lwt                    1.0.2  An OCaml library for HTTP clients and serve
cohttp-lwt-unix               1.0.2  An OCaml library for HTTP clients and serve

$ ocamlbuild -pkg cohttp-lwt-unix server_example.native
+ /home/thegaram/.opam/4.06.1/bin/ocamlopt.opt -I /home/thegaram/.opam/4.06.1/lib/astring -I /home/thegaram/.opam/4.06.1/lib/base -I /home/thegaram/.opam/4.06.1/lib/base/caml -I /home/thegaram/.opam/4.06.1/lib/base/shadow_stdlib -I /home/thegaram/.opam/4.06.1/lib/base64 -I /home/thegaram/.opam/4.06.1/lib/bytes -I /home/thegaram/.opam/4.06.1/lib/cohttp -I /home/thegaram/.opam/4.06.1/lib/cohttp-lwt -I /home/thegaram/.opam/4.06.1/lib/cohttp-lwt-unix -I /home/thegaram/.opam/4.06.1/lib/conduit -I /home/thegaram/.opam/4.06.1/lib/conduit-lwt -I /home/thegaram/.opam/4.06.1/lib/conduit-lwt-unix -I /home/thegaram/.opam/4.06.1/lib/fieldslib -I /home/thegaram/.opam/4.06.1/lib/fmt -I /home/thegaram/.opam/4.06.1/lib/ipaddr -I /home/thegaram/.opam/4.06.1/lib/ipaddr/unix -I /home/thegaram/.opam/4.06.1/lib/logs -I /home/thegaram/.opam/4.06.1/lib/lwt -I /home/thegaram/.opam/4.06.1/lib/magic-mime -I /home/thegaram/.opam/4.06.1/lib/ocaml -I /home/thegaram/.opam/4.06.1/lib/re -I /home/thegaram/.opam/4.06.1/lib/re/emacs -I /home/thegaram/.opam/4.06.1/lib/re/posix -I /home/thegaram/.opam/4.06.1/lib/result -I /home/thegaram/.opam/4.06.1/lib/sexplib -I /home/thegaram/.opam/4.06.1/lib/sexplib/0 -I /home/thegaram/.opam/4.06.1/lib/stringext -I /home/thegaram/.opam/4.06.1/lib/uchar -I /home/thegaram/.opam/4.06.1/lib/uri -I /home/thegaram/.opam/4.06.1/lib/uri/services /home/thegaram/.opam/4.06.1/lib/base64/base64.cmxa /home/thegaram/.opam/4.06.1/lib/base/caml/caml.cmxa /home/thegaram/.opam/4.06.1/lib/base/shadow_stdlib/shadow_stdlib.cmxa /home/thegaram/.opam/4.06.1/lib/sexplib/0/sexplib0.cmxa /home/thegaram/.opam/4.06.1/lib/base/base.cmxa /home/thegaram/.opam/4.06.1/lib/fieldslib/fieldslib.cmxa /home/thegaram/.opam/4.06.1/lib/re/re.cmxa /home/thegaram/.opam/4.06.1/lib/re/emacs/re_emacs.cmxa /home/thegaram/.opam/4.06.1/lib/ocaml/unix.cmxa /home/thegaram/.opam/4.06.1/lib/ocaml/bigarray.cmxa /home/thegaram/.opam/4.06.1/lib/sexplib/sexplib.cmxa /home/thegaram/.opam/4.06.1/lib/stringext/stringext.cmxa /home/thegaram/.opam/4.06.1/lib/re/posix/re_posix.cmxa /home/thegaram/.opam/4.06.1/lib/uri/uri.cmxa /home/thegaram/.opam/4.06.1/lib/cohttp/cohttp.cmxa /home/thegaram/.opam/4.06.1/lib/result/result.cmxa /home/thegaram/.opam/4.06.1/lib/lwt/lwt.cmxa /home/thegaram/.opam/4.06.1/lib/cohttp-lwt/cohttp_lwt.cmxa /home/thegaram/.opam/4.06.1/lib/astring/astring.cmxa /home/thegaram/.opam/4.06.1/lib/ipaddr/ipaddr.cmxa /home/thegaram/.opam/4.06.1/lib/conduit/conduit.cmxa /home/thegaram/.opam/4.06.1/lib/conduit-lwt/conduit_lwt.cmxa /home/thegaram/.opam/4.06.1/lib/ipaddr/unix/ipaddr_unix.cmxa /home/thegaram/.opam/4.06.1/lib/logs/logs.cmxa /home/thegaram/.opam/4.06.1/lib/lwt/lwt_log.cmxa /home/thegaram/.opam/4.06.1/lib/lwt/lwt_unix.cmxa /home/thegaram/.opam/4.06.1/lib/uri/services/uri_services.cmxa /home/thegaram/.opam/4.06.1/lib/conduit-lwt-unix/conduit_lwt_unix.cmxa /home/thegaram/.opam/4.06.1/lib/fmt/fmt.cmxa /home/thegaram/.opam/4.06.1/lib/logs/logs_lwt.cmxa /home/thegaram/.opam/4.06.1/lib/magic-mime/magic_mime_library.cmxa /home/thegaram/.opam/4.06.1/lib/cohttp-lwt-unix/cohttp_lwt_unix.cmxa server_example.cmx -o server_example.native
File "_none_", line 1:
Error: No implementations provided for the following modules:
         Cohttp_lwt_unix__ referenced from server_example.cmx
         Cohttp_lwt__ referenced from server_example.cmx
         Cohttp__ referenced from server_example.cmx
Command exited with code 2.
Hint: Recursive traversal of subdirectories was not enabled for this build,
  as the working directory does not look like an ocamlbuild project (no
  '_tags' or 'myocamlbuild.ml' file). If you have modules in subdirectories,
  you should add the option "-r" or create an empty '_tags' file.
  
  To enable recursive traversal for some subdirectories only, you can use the
  following '_tags' file:
  
      true: -traverse
      <dir1> or <dir2>: traverse
      
Compilation unsuccessful after building 4 targets (3 cached) in 00:00:00.
```

I've tried with multiple compiler versions, all fail with this error. I haven't experienced this problem on a Mac. The problematic Cohttp packages seem to be properly installed. Any help is appreciated.

**Comment by Thegaram** [comment]
#436 is related but doesn't seem to help.

**Comment by rgrinberg** [comment]
You need to use jbuilder beta17 and then recompile cohttp. Sorry about this. It will be fixed in jbuilder 18.1

**Comment by Thegaram** [comment]
Yep, it works now, thanks!
